akka-stream/src/main/scala/akka/stream/impl/FanoutProcessor.scala

```diff
@@ -121,14 +121,6 @@ import org.reactivestreams.Subscriber
     primaryOutputs.enqueueOutputElement(primaryInputs.dequeueInputElement())
   }
 
-  override def fail(e: Throwable): Unit = {
-    if (settings.debugLogging)
-      log.debug("fail due to: {}", e.getMessage)
-    primaryInputs.cancel()
-    primaryOutputs.error(e)
-    // Stopping will happen after flush
-  }
```

Comment:
always scary to change/remove something that someone else has consciously added and even commented, but obviously there is a problem and no other test fails

Reply:
Yeah, I wonder if it was true at some point but other things changed around it perhaps. Or some assumption was wrong.

Reply:
With all due respect, I’m a bit surprised at the questions being raised here. If you take the comment at face value, you can easily disprove the claim. If it did happen to be that the behavior was changed due to an unintended consequence, the idea that the corner case would not have been documented seems flawed, unless you assume the second author was less diligent than the first.

For me, this issue caused the process to run out of memory in a non obvious and way, with no workaround without a redesign. While I can appreciate the desire to be attentive to risks that can lead to unexpected regressions, but arbitrary risk aversion has costs. I think you guys created a great product with easy to grasp abstracions that clearly show a respect for the user experience, and I appreciate the thought and consideration you and your team put into it.

My issue is though that by investing a little more time in learning rxjava2 or monix, there are far less surprises.